src/util/moc_included_test.cpp

```diff
@@ -0,0 +1,7 @@
+#include "../mocs_compilation.cpp"
+
+// QT_VERSION will be defined by any moc_<header_base>.cpp file included from mocs_compilation.cpp
+// It is empty in case all moc files are included
+#ifdef QT_VERSION
+#error "Not all moc_<header_base>.cpp files included from their cpp sources. This is a Mixxx requirement to speed up the build process. See https://cmake.org/cmake/help/latest/prop_tgt/AUTOMOC.html for details."
```

Comment:
The link doesn't really explain why this makes sense. It just explains the CMake `moc` integration. Thats why I asked the other questions previously since the benefit just isn't apparent by reading the code and comments.

Reply:
The link in the errror message was ment to help resolving the error when it occurs. 
Do you wish to improve it?

Reply:
yes, I think someone encountering this error doesn't really what to do to make it go away. Neither do so I struggle to find a better message. I think adding something like `please ensure to add #include "moc_<header_base>.cpp" in your all files with QObject-based classes`. is that correct? I'd then move the "This is a Mixxx requirement to speed up the build process. See https://cmake.org/cmake/help/latest/prop_tgt/AUTOMOC.html for details." into a plain comment.